server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -484,11 +485,66 @@ void executeRequest(
                 }
             }
         });
+        final SearchSourceBuilder source = original.source();
+        if (shouldOpenPIT(source)) {
+            openPIT(original, searchService.getDefaultKeepAliveInMillis(), listener.delegateFailureAndWrap((delegate, resp) -> {
+                source.pointInTimeBuilder(new PointInTimeBuilder(resp.getPointInTimeId()).setKeepAlive(TimeValue.MINUS_ONE));
```

Comment:
What is the meaning of setting keepAlive to -1 ms?

Reply:
In this PR it means that the pit id is not returned in the response.

Reply:
May be worth addign a comment about it.

Reply:
https://github.com/elastic/elasticsearch/pull/110482/commits/a370caca0fb026890e6a8a893591658fec6a6fb6